src/plots/elexon_plots.py

```diff
@@ -0,0 +1,65 @@
+import pandas as pd
+from datetime import datetime, date, timedelta
+from plotly import graph_objects as go
+import streamlit as st
+
+def fetch_forecast_data(api_func, start_date, end_date, process_type):
+    try:
```

Comment:
Cna you add function descriptions and say what the parameters are.

Could you add type hinting please too,